workbench/test/selenium_test.py

```diff
@@ -19,3 +23,15 @@ def setUp(self):
         # Clear the in-memory key value store, the usage store, and whatever
         # else needs to be cleared and re-initialized.
         reset_global_state()
+
+    @contextmanager
+    def wait_for_page_load(self, timeout=30):
+        """
+        Uses Selenium's built-in "staleness" hook to wait until the page has
+        loaded. For use when clicking a link to ensure that the new page
+        has loaded before selecting elements.
+        """
+        old_page = self.browser.find_element_by_tag_name('html')
+        yield WebDriverWait(self.browser, timeout).until(
+            staleness_of(old_page)
+        )
```

Comment:
Ah.  Looking at that link, I think you had a slight misinterpretation of their code.  They had:

    yield
    WebDriverWait(...)

meaning they waited after coming back from the yield.

Your version had

    yield WebDriverWait(...)

meaning execute the wait, and then yield the response of that function.  So the yielding happened on the wrong side of the weight.   (My version matches that of the book, but actually yields out the value of the old element, which you probably don't need, but could on occasion be useful.  Also, my try/finally is probably unnecessary and maybe even wrong.  If an exception occurs, we don't need to wait for the page to go stale anymore.